lib/letsencrypt.js

```diff
@@ -1,215 +1,104 @@
-                                                });
-                                                for (var r = 0; r < running.length; r++) {
-                                                    running[r](null, hostTls);
-                                                }
-                                            } else {
-                                                 for (var r = 0; r < running.length; r++) {
-                                                     running[r](null, defaultTls);
-                                                 }
-                                            }
-                                            running = true;
-                                        });
-                                    } else {
-                                        if (running === true) {
-                                            cb(null, defaultTls);
-                                        } else {
-                                            running.push(cb);
-                                        }
-                                    }
-                                    */
-                                }
-                            },
-                            err => {
-                                err && log.error(err);
-                                cb(null, defaultTls);
-                            });
-                        } else {
-                            cb(null, hostTls);
-                        }
+                // prepare domains
+                if (typeof leSettings.domains === 'string') {
+                    leSettings.domains = leSettings.domains.split(',');
+                    for (let d = leSettings.domains.length - 1; d >= 0; d--) {
+                        leSettings.domains[d] = leSettings.domains[d].trim();
+                        // delete empty
+                        !leSettings.domains[d] && leSettings.domains.splice(d, 1);
                     }
-                } else {
-                    cb(null, defaultTls);
                 }
-            };
+                // TODO? Tried adding sites in the gl create but that didn't work so is below with gl.add.
+
+                const pkg = require('../package.json');
+                const gl = require('greenlock').create({
+                    packageRoot: __dirname,
+                    configDir: leDir,
+                    maintainerEmail: leSettings.email,
+                    cluster: false,
+                    packageAgent: pkg.name + '/' + pkg.version,
+                    notify: function (event, details) {
+                        // TODO: separate out errors, etc.
+                        log.debug(`Greenlock notify ${event}: ${details}`);
+                    }
+                });
+                gl.manager.defaults({
+                    agreeToTerms: true,
+                    subscriberEmail: leSettings.email
+                }).then(() => {
```

Comment:
Is this a promise that is returned? If so, we should await it. If not, is there an error condition we should look for?

Reply:
I did think about that, but given this is itself inside a Promise resolve function adding an `await` in there won't work. Sure, we could split this into multiple functions, and normally I like moving blocks like that, but in this case not sure there is much benefit.

Reply:
But a `.catch()` would be good to handle error cases. As we won't resovle the original promise if this fails (if it can fail).

Reply:
So what do you want to do inside the `.catch()`? Log an error and reject the Promise? I don't see the point when the caller already needs to catch any errors themselves.

Reply:
Yes, just reject the Promise. I hate promises that never "end". Like that the caller of the entire method at least has a chance to handle this.

Reply:
OK, done in https://github.com/ioBroker/ioBroker.js-controller/pull/1150/commits/47cc6b6a8ce8cba40d96f9bd12976a6457f910e8